src/Bicep.LangServer/Handlers/ExternalSourceReference.cs

```diff
@@ -17,11 +18,22 @@ namespace Bicep.LanguageServer.Handlers
     /// </summary>
     public class ExternalSourceReference
     {
-        // The title to display for the document,
-        //   e.g. "br:myregistry.azurecr.io/myrepo/module/v1/main.json (module:v1)" or something similar.
+        // e.g. matches <cache>/br/mcr.microsoft.com/bicep$storage$storage-account/1.0.1$/main.json
+        private Regex externalModulePath = new("""
+            \<cache\>\/br\/
+            .*
+            \$(?<repoName>[^\/\$]+)
+            \/(?<tag>[^\/\$]+)\$[^\/\$]*
+            \/(?<filename>[^\/]+)$            
+            """,
+
+            RegexOptions.IgnorePatternWhitespace);
```

Comment:
Might be worth using regex [source generator](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators) instead.

Reply:
c5df11e3b